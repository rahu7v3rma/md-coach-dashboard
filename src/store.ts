import { configureStore } from '@reduxjs/toolkit';

import clientReducer from './reducers/client';
import groupReducer from './reducers/group';
import notificationReducer from './reducers/notification';
import userReducer from './reducers/user';

// define store options as a variable so we can export it for tests to use
const storeOptions: any = {
    reducer: {
        user: userReducer,
        client: clientReducer,
        notification: notificationReducer,
        group: groupReducer
    }
};

const store = configureStore(storeOptions);

export default store;

// infer the `RootState` type from the store
export type RootState = ReturnType<typeof store.getState>;

// infer the `AppDispatch` type from the store so thunk actions can return promises
export type AppDispatch = typeof store.dispatch;

export const test = {
    storeOptions
};
