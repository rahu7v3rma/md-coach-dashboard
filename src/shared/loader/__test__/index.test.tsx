import 'jest-styled-components';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import Loader, { StyledLoader } from '..';
import store from '../../../store';

describe('Loader', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('Loader snapshot', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <Loader />
                </BrowserRouter>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });

    it('Loader renders without errors', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <Loader />
                </BrowserRouter>
            </Provider>
        );
        expect(tree.root).toBeTruthy();
    });

    it('should have the correct CSS animations', async () => {
        const tree = create(<StyledLoader />).toJSON();
        expect(tree).toHaveStyleRule('animation', 'rotate 2s linear infinite');
    });

    it('should have the correct circle attributes', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <Loader />
                </BrowserRouter>
            </Provider>
        );
        const circle = tree.root.findByProps({
            className: 'path'
        }).props;
        expect(circle.cx).toBe('25');
        expect(circle.cy).toBe('25');
        expect(circle.r).toBe('20');
        expect(circle.fill).toBe('none');
        expect(circle.strokeWidth).toBe('4');
    });

    it('should have the correct SVG attributes', async () => {
        const tree = create(<StyledLoader />).toJSON();
        expect(tree).toHaveStyleRule('animation', 'rotate 2s linear infinite');
        expect(tree).toHaveStyleRule('width', '25px');
        expect(tree).toHaveStyleRule('height', '25px');
    });
});
