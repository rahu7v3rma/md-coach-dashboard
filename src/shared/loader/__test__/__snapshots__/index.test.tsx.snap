// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Loader snapshot 1`] = `
.c0 {
  -webkit-animation: rotate 2s linear infinite;
  animation: rotate 2s linear infinite;
  width: 25px;
  height: 25px;
}

.c0 .path {
  stroke: #5652bf;
  stroke-linecap: round;
  -webkit-animation: dash 1.5s ease-in-out infinite;
  animation: dash 1.5s ease-in-out infinite;
}

<svg
  className="c0"
  viewBox="0 0 50 50"
>
  <circle
    className="path"
    cx="25"
    cy="25"
    fill="none"
    r="20"
    strokeWidth="4"
  />
</svg>
`;
